plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -2,94 +2,97 @@
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-
 DOCUMENTATION = '''
 ---
 module: newrelic_deployment
-author: "Matt Coddington (@mcodd)"
+author:
+    - "Davinder Pal (@116davinder)"
+    - "Matt Coddington (@mcodd)"
 short_description: Notify newrelic about app deployments
 description:
-   - Notify newrelic about app deployments (see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/deployment-notifications#api)
+   - Notify newrelic about app deployments
+        (U(https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-deployments)).
 options:
   token:
-    type: str
     description:
       - API token, to place in the x-api-key header.
     required: true
-  app_name:
     type: str
+  app_name:
     description:
-      - (one of app_name or application_id are required) The value of app_name in the newrelic.yml file used by the application
+      - The value of C(app_name) in the C(newrelic.yml) file used by the application.
+      - Exactly one of I(app_name) or I(application_id) is required.
     required: false
-  application_id:
     type: str
+  application_id:
     description:
-      - (one of app_name or application_id are required) The application id, found in the URL when viewing the application in RPM
+      - See U(https://rpm.newrelic.com/api/explore/applications/list).
+      - Exactly one of I(app_name) or I(application_id) is required.
     required: false
-  changelog:
     type: str
+  changelog:
     description:
-      - A list of changes for this deployment
+      - A list of changes for this deployment.
     required: false
-  description:
     type: str
+  description:
     description:
-      - Text annotation for the deployment - notes for you
+      - Text annotation for the deployment - notes for you.
     required: false
-  revision:
     type: str
+  revision:
     description:
-      - A revision number (e.g., git commit SHA)
-    required: false
-  user:
+      - A revision number (for example, git commit SHA).
+    required: true
     type: str
-    description:
-      - The name of the user/process that triggered this deployment
-    required: false
   appname:
     type: str
     description:
-      - Name of the application
+      - (deprecated) Name of the application
     required: false
   environment:
     type: str
     description:
-      - The environment for this deployment
+      - (deprecated) The environment for this deployment.
     required: false
   validate_certs:
     description:
-      - If C(no), SSL certificates will not be validated. This should only be used
+      - (deprecated) If C(no), SSL certificates will not be validated. This should only be used
         on personally controlled sites using self-signed certificates.
```

Comment:
Why is this option deprecated?

Reply:
may be this will refresh your memory:
https://github.com/ansible-collections/community.general/pull/876#issuecomment-691019903